### PR TITLE
[receiver/hostmetrics] Unify feature gates for `direction` removal

### PR DESCRIPTION
**Pull request by dmitryax**
This change replaces set of independent feature gates for transition from the metrics with `direction` attribute with two unified feature gates that are applied on all the metrics reported by the hostmetrics receiver.
- `receiver.hostmetricsreceiver.emitMetricsWithDirectionAttribute`
- `receiver.hostmetricsreceiver.emitMetricsWithoutDirectionAttribute`

Also this introduces the following changes to the transition process:
1. I shifts the transition schedule one release forward since disc scraper is not yet ready to emit the new metrics
2. It removes the warnings. They will be introduced in 0.56.0 when all the scrapers are aligned with the schedule.
3. It introduces an option to emit both the deprecated and the new metrics. This can be useful for users during the transition period.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/11815

### CODE REVIEW

**Comment by dmitryax** [comment]
@codeboten PTAL and let me know what do you think. I would like to include it in the 0.55.0 release

**Comment by tigrannajaryan** [comment]
@open-telemetry/collector-contrib-approvers please review/merge or remove the `release:blocker` label.

**Review by mx-psi — APPROVED** [review]
This looks good to me, but would prefer if @codeboten had a look too if he has more context

**Review by codeboten — COMMENTED** [review]
Just one question and a comment, this consolidation makes sense to me. However it might be worth capturing here that it wont apply to the `disk` scraper just yet as that change hasn't been implemented.

File: receiver/hostmetricsreceiver/README.md
Comment (dmitryax) [reply receiver/hostmetricsreceiver/README.md]: fixed

File: receiver/hostmetricsreceiver/README.md
Comment (dmitryax) [reply receiver/hostmetricsreceiver/README.md]: I added another step for the current state

**Review by codeboten — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
Please merge once the build passes. I will wait for it and will start the contrib release procedure.

**Comment by dmitryax** [comment]
@tigrannajaryan, merged

**Pull request merged**
